tests/specs/lemmas.k

```diff
@@ -5,20 +5,42 @@ module LEMMAS
     imports LEMMAS-JAVA
     imports LEMMAS-HASKELL
 
+    rule WS ++ .ByteArray => WS  [simplification]
+    rule .ByteArray ++ WS => WS  [simplification]
+
+    //Byte array store operations sorting. Slices with lower index first. Helps applying rule below.
+    rule MEM [ N := BUF:ByteArray ] [ M := BUF' ] => MEM [ M := BUF' ] [ N := BUF ]
+      requires N >Int M andBool N -Int M >=Int #sizeByteArray(BUF')                                               [simplification]
+
+    rule M [ N := BUF ] [ N := BUF' ] => M [ N := BUF' ] requires #sizeByteArray(BUF) ==Int #sizeByteArray(BUF')  [simplification]
+
+    rule BUF [ L .. W ] => .ByteArray requires W <=Int 0                    [simplification]
```

Comment:
Is this one needed anymore with the new rules for `_[_.._]`?

Reply:
It is still needed for case `W ==Int 0`. I simplified the rule.